sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -104,6 +104,7 @@ class Analyzer(
       ResolveAggregateFunctions ::
       TimeWindowing ::
       ResolveInlineTables ::
+      ResolveTimeZone ::
```

Comment:
@hvanhovell Thank you for your suggestion.
I overrode `resolved` of `TimeZoneAwareExpression`s, then we needed to add `ResolveTimeZone` to `Resolution` batch, but I found we don't need to worry about the resolution because to have the timezone or not doesn't affect the resolution and we only need to run once now.